downloader/src/main/kotlin/Downloader.kt

```diff
@@ -53,19 +53,45 @@ class Downloader {
      * The choice of data entities to download.
      */
     enum class DataEntity {
+        /**
+         * Identifier for package entities.
+         */
         PACKAGES,
+
+        /**
+         * Identifier for project entities.
+         */
         PROJECTS;
     }
 
     /**
-     * This class describes what was downloaded by [download] to the [downloadDirectory] or if any exception occurred.
-     * Either [sourceArtifact] or [vcsInfo] is set to a non-null value. The download was started at [dateTime].
+     * This class describes what was downloaded by any of the download functions.
      */
     data class DownloadResult(
+        /**
+         * The date and time when the download was started.
+         */
         val dateTime: Instant,
+
+        /**
+         * The directory to which files were downloaded.
+         */
         val downloadDirectory: File,
+
+        /**
+         * The source artifact that was downloaded, or null if the download was performed from a VCS.
```

Comment:
I would link the properties in the docs to better show the mutual exclusivity.
`...from a [VCS][vcsInfo].`

Reply:
Done.